bpo-29753: fix merging packed bitfields in ctypes struct/union (#19850)
python/cpython: Misc/NEWS.d/next/Library/2020-05-02-01-01-30.bpo-29753.n2M-AF.rst

> Fixes packed bitfields population.

Reviewer: This shows up in the changelog with no other context other than a link to the issue ticket, so you should elaborate as best you can in 1-2 sentences what was broken before and what is now fixed.

> Fixes packed bitfields not being calculated properly in some cases.
> Fixes packed bitfileds not shrinking the first item.

Reviewer 2: I would re-write this to give users more context. Think about the person reading the NEWS for Python 3.10. This notice doesn't explain which bitfields were not being calculated properly.

```suggestion
In ctypes, now packed bitfields are calculated properly and packed bitfields avoid shrinking the first item.
```

Author: Thanks.

> packed bitfields avoid shrinking the first item

It's the opposite :stuck_out_tongue: They are now shrank, as they should. I'll take this as a failure of my initial description.

Fixes packed bitfields not being calculated properly in some cases (see

> bpo-29753).

Reviewer 3: Please do not mention the issue number in the news entry (it is automatically added).

> Fixes packed bitfileds not being packed when created.

Reviewer 3: Please fix the spelling in this sentence.

In my opinion, the news entry is rather vague currently and could use more detail.

Reviewer: Yeah, I agree, this could use more detail. Keep in mind that this will be displayed with effectively no context other than the bpo issue, and bpo issues are often hard to read because there's a lot of back-and-forth trying to figure out the cause of the problem, what the solution is, etc. It's best to give a 1-2 sentence summary of what the problem is and what you changed to fix it.
